osidb/serializer.py

```diff
@@ -919,19 +919,40 @@ class Meta:
         abstract = True
 
 
-class AffectCVSSSerializer(
+class AbstractCVSSSerializer(
     ACLMixinSerializer,
     AlertMixinSerializer,
     BugzillaSyncMixinSerializer,
     IncludeExcludeFieldsMixin,
     TrackingMixinSerializer,
+):
+    """Abstract serializer for FlawCVSS and AffectCVSS serializer"""
+
+    cvss_version = serializers.ChoiceField(
+        choices=FlawCVSS.CVSSVersion, source="version"
+    )
+    score = serializers.FloatField(read_only=True)
+    comment = serializers.CharField(required=False, allow_null=True, allow_blank=True)
+
+    class Meta:
+        abstract = True
+        fields = (
+            # Also add "affect" or "flaw" for AffectCVSS or FlawCVSS
+            ["comment", "cvss_version", "issuer", "score", "uuid", "vector"]
+            + ACLMixinSerializer.Meta.fields
+            + AlertMixinSerializer.Meta.fields
+            + TrackingMixinSerializer.Meta.fields
+        )
```

Comment:
I am not sure if the current abstract class makes much sense. Can the changes be made directly in the respective classes?

Reply:
The changes certainly can be made directly in the respective classes, but the dichotomy that the base CVSS abstract model is used in two different models that are used in two different serializers was lost on me and resulted in a day of lost time.

Given that and after a chat with @JakubFrejlach I deemed it worthwhile to have the common abstract serializer to make it non-confusing when the next change on CVSS occurs.

Reply:
I briefly checked the DRF documentation (https://www.django-rest-framework.org/api-guide/serializers/), and there seem to be several options on how to solve the "None to empty string" issue, e.g.:
1. Overriding `.to_representation()` and `.to_internal_value()` methods in a serializer for custom serialization/deserialization (i.e. converting None to an empty string in serializer)
2. Creating a custom char field and automatically convert None to an empty string

If I understand it correctly, currently, the serializer enables both None and empty string, and the conversion is done in models via `NullStrFieldsMixin`. But in future, we should handle the conversion directly in a serializer and then save either string or empty string in models.

Reply:
This may be a good suggestion for the future but for today we cannot afford any bigger changes. Let us only focus on making this working properly and let us leave the improvements for the next time.